Source/Urho3D/Network/Network.h

```diff
@@ -51,16 +52,20 @@ class URHO3D_API Network : public Object, public kNet::IMessageHandler, public k
     /// Destruct.
     ~Network() override;
 
-    /// Handle a kNet message from either a client or the server.
-    void HandleMessage
-        (kNet::MessageConnection* source, kNet::packet_id_t packetId, kNet::message_id_t msgId, const char* data, size_t numBytes) override;
-    /// Compute the content ID for a message.
-    u32 ComputeContentID(kNet::message_id_t msgId, const char* data, size_t numBytes) override;
-    /// Handle a new client connection.
-    void NewConnectionEstablished(kNet::MessageConnection* connection) override;
-    /// Handle a client disconnection.
-    void ClientDisconnected(kNet::MessageConnection* connection) override;
 
+    /// Handle an inbound message.
+    virtual void HandleMessage(const SLNet::AddressOrGUID& source, int packetID, int msgID, const char* data, size_t numBytes);

```

Comment:
Why virtual?

Reply:
I believe this is an artifact left from kNet implementation. Network previously implemented kNet listener classes. Same for NewConnectionEstablished and ClientDisconnected methods